test/unit/gateways/flex_charge_test.rb

```diff
@@ -106,6 +119,34 @@ def test_successful_purchase
     assert response.test?
   end
 
+  def test_successful_purchase_three_ds_global
+    response = stub_comms do
+      @gateway.purchase(@amount, @credit_card, @three_d_secure_options)
+    end.respond_with(successful_access_token_response, successful_purchase_response)
+    assert_success response
+    assert_equal 'ca7bb327-a750-412d-a9c3-050d72b3f0c5', response.authorization
+    assert response.test?
+  end
+
+  def test_succeful_request_with_three_ds_global
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @three_d_secure_options)
+    end.check_request do |endpoint, data, _headers|
+      if /evaluate/.match?(endpoint)
+        request = JSON.parse(data)
+        assert_equal request['threeDSecure']['EcommerceIndicator'], @three_d_secure_options[:three_d_secure][:eci]
```

Comment:
💭 Opinion.
There are several repetitions in your code, especially in accessing @three_d_secure_options[:three_d_secure], and request['threeDSecure']. You can simplify this by assigning it to a variable at the beginning of your method.

Reply:
thanks, updated